src/vaults/interfaces/IVaultRouter.sol

```diff
@@ -151,15 +151,21 @@ interface IVaultRouter is IMulticall, IRecoverable {
     /// @dev    This adds a mandatory prepayment for all the costs that will incur during the transaction.
     ///         The caller must call `VaultRouter.estimate` to get estimates how much the deposit will cost.
     ///
+    /// @param  centrifugeId Chain to where transfer the shares
     /// @param  vault The vault for the corresponding share class token
-    /// @param  chainId Check `IPoolManager.transferShares.destinationId`
     /// @param  receiver Check `IPoolManager.transferShares.receiver`
     /// @param  amount Check `IPoolManager.transferShares.amount`
-    function transferShares(address vault, uint16 chainId, bytes32 receiver, uint128 amount) external payable;
+    function transferShares(uint16 centrifugeId, address vault, bytes32 receiver, uint128 amount) external payable;
 
     /// @notice This is a more friendly version where the receiver is and EVM address
     /// @dev    The receiver address is padded to 32 bytes internally
-    function transferShares(address vault, uint16 chainId, address receiver, uint128 amount) external payable;
+    function transferShares(uint16 centrifugeId, address vault, address receiver, uint128 amount) external payable;
+
+    /// @notice Register an asset to be used in a pool existing in centrifugeId
+    /// @param centrifugeId Where the asset will be registered
+    /// @param asset If tokenId == 0, an ERC20 address, if tokenId != 0, and ERC6909 address
+    /// @param tokenId An ERC6909 tokenId
+    function registerAsset(uint16 centrifugeId, address asset, uint256 tokenId) external payable;
```

Comment:
Curious why this was added to the `VaultRouter` interace. Is the current `PoolManager` exposure insufficient?

Reply:
Without exposing it, we're not able to:
1. Pay for the underlying message
2. batch it

Reply:
Actually we should make it auth to avoid any user to call it and drain the pool0 funds (funds intented for unrelated pool method). Will add that in follow up PR

Reply:
Makes sense! Users can still pay for that in the router, right? Great catch!

Reply:
Will do there, yes